sycl/include/sycl/ext/intel/experimental/esimd/detail/type_format.hpp

```diff
@@ -0,0 +1,84 @@
+//==-------------- types.hpp - DPC++ Explicit SIMD API ---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Meta-functions to compute compile-time element type of a simd_view resulting
+// from format operations.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/intel/experimental/esimd/detail/types.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace __SEIEED {
+
+template <typename BaseTy, typename EltTy> struct compute_format_type;
+
+template <typename Ty, int N, typename EltTy> struct compute_format_type_impl {
+  static constexpr int Size = sizeof(Ty) * N / sizeof(EltTy);
+  static constexpr int Stride = 1;
+  using type = region1d_t<EltTy, Size, Stride>;
+};
+
+template <typename Ty, int N, typename EltTy,
+          template <typename, int> class SimdT>
+struct compute_format_type<SimdT<Ty, N>, EltTy>
+    : compute_format_type_impl<Ty, N, EltTy> {};
+
```

Comment:
specialization for <simd_obj_impl<Ty, N, SimdT>, EltTy> was removed - not supposed to be used as argument